Modules/Data/Defense.lua

```diff
@@ -42,29 +55,110 @@ function _Defense:GetCritReduction()
+                spellCritReduction = spellCritReduction + 2
+            end
+        end
+    elseif classId == Data.ROGUE then
+        if IsPlayerSpell(30893) then -- Sleight of Hand 2/2
+            meleeCritReduction = meleeCritReduction + 2
+            rangedCritReduction = rangedCritReduction + 2
+        elseif IsPlayerSpell(30892) then -- Sleight of Hand 1/2
+            meleeCritReduction = meleeCritReduction + 1
+            rangedCritReduction = rangedCritReduction + 1
+        end
+    elseif classId == Data.WARLOCK then
+        if ECS.IsTBC or ECS.IsWotlk then
+            if IsPlayerSpell(30321) then -- Demonic Resilience 3/3
+                meleeCritReduction = meleeCritReduction + 3
+                spellCritReduction = spellCritReduction + 3
+            elseif IsPlayerSpell(30320) then --  Demonic Resilience 2/3
+                meleeCritReduction = meleeCritReduction + 2
+                spellCritReduction = spellCritReduction + 2
+            elseif IsPlayerSpell(30319) then -- Demonic Resilience 1/3
+                meleeCritReduction = meleeCritReduction + 1
+                spellCritReduction = spellCritReduction + 1
+            end
+        end
+        if ECS.IsWotlk then
+            if C_UnitAuras.GetPlayerAuraBySpellID(47241) then
+                meleeCritReduction = meleeCritReduction + 6 -- metamorphosis
+            end
+        end
+    end
 
-    if critImmunity < 0 then
-        critImmunity = 0
+    if ECS.IsSoD then
+        if classId == Data.DRUID or classId == Data.ROGUE then
+            local chestRune = DataUtils.GetRuneForEquipSlot(Utils.CHAR_EQUIP_SLOTS.Chest)
+            if chestRune and (chestRune == 6710 or chestRune == 6972) then
+                meleeCritReduction = meleeCritReduction + 6 -- survival of the fittest / Just a Flesh Wound
+            end
+        end
+    elseif ECS.IsWotlk then
+        local wintersChill = C_UnitAuras.GetPlayerAuraBySpellID(12579)
+        if wintersChill then
+            spellCritReduction = spellCritReduction - 1 * wintersChill.applications -- Winter's Chill
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(22959) then
+            spellCritReduction = spellCritReduction - 5 -- Improved Scorch
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(17800) then
+            spellCritReduction = spellCritReduction - 5 -- Shadow Mastery
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(17799) then
+            spellCritReduction = spellCritReduction - 4 -- Shadow Mastery
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(17798) then
+            spellCritReduction = spellCritReduction - 2 -- Shadow Mastery
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(17797) then
+            spellCritReduction = spellCritReduction - 3 -- Shadow Mastery
+        end
+        if C_UnitAuras.GetPlayerAuraBySpellID(17794) then
+            spellCritReduction = spellCritReduction - 1 -- Shadow Mastery
+        end
```

Comment:
Same as with the others, this is only valid in PvP cases, right?

Reply:
pvp or you get your own debuff spell reflected on yourself